### PR TITLE
Added Detect Cycle in Undirected Graph

### PR DESCRIPTION
**Pull request by akashksinghal**
Added Detect Cycle in Undirected Graph

### CODE REVIEW

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+			return true; 
+	return false; 
+} 
+
+// Driver program to test methods of graph class 
+int main() 
+{ 
+	int V = 4; 
+	vector<int> adj[V]; 
+	addEdge(adj, 0, 1);
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: make it dynamic

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+#include <bits/stdc++.h> 
+using namespace std; 
+
+void addEdge(vector<int> adj[], int u, int v) 
+{ 
+	adj[u].push_back(v); 
+	adj[v].push_back(u); 
+} 
+
+bool isCyclicConntected(vector<int> adj[], int s,
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: What is s, V?

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	// Mark the current node as visited and enqueue it 
+	visited[s] = true; 
+	q.push(s); 
+
+	while (!q.empty()) { 
+
+		// Dequeue a vertex from queue and print it 
+		int u = q.front(); 
+		q.pop();
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: What is q?

Comment (akashksinghal) [reply Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: q is a queue. It is stated in comment at line 16

**Review by akashksinghal — COMMENTED** [review]
Updated Required Changes

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp (hunk 1 above)
Comment (akashksinghal) [reply Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: Done

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp (hunk 2 above)
Comment (akashksinghal) [reply Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: Changed the notation of s, and commented what V means

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp [hunk 4]
```diff
@@ -44,24 +45,32 @@ bool isCyclicDisconntected(vector<int> adj[], int V)
...
 			return true; 
 	return false; 
 } 
 
 // Driver program to test methods of graph class 
 int main() 
 { 
-	int V = 4; 
+	cout<<"enter the number of vertices : ";
+	int V ;//V stands for Number of Verticies
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: change the notation

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp [hunk 5]
```diff
@@ -8,27 +8,28 @@ void addEdge(vector<int> adj[], int u, int v)
...
+	vector<int> parent(NoOfVertices, -1); 
 
 	// Create a queue for BFS 
 	queue<int> q; 
 
 	// Mark the current node as visited and enqueue it 
-	visited[s] = true; 
-	q.push(s); 
+	visited[VertexToBeVisited] = true; 
+	q.push(VertexToBeVisited);
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: let q be queue

**Review by akashksinghal — COMMENTED** [review]
updated as adviced

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp (hunk 4 above)
Comment (akashksinghal) [reply Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: updated

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp [hunk 6]
```diff
@@ -55,11 +55,11 @@ bool isCyclicDisconntected(vector<int> adj[], int V)
 int main() 
 { 
 	cout<<"enter the number of vertices : ";
-	int V ;//V stands for Number of Verticies
-	cin>>V;
+	int NoOfVertices ;
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: snake case variable not camel case

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp [hunk 7]
```diff
@@ -14,24 +14,24 @@ bool isCyclicConntected(vector<int> adj[], int VertexToBeVisited,int NoOfVertice
...
-	queue<int> q; 
+	queue<int> Queue; 
 
 	// Mark the current node as visited and enqueue it 
 	visited[VertexToBeVisited] = true; 
-	q.push(VertexToBeVisited); 
+	Queue.push(VertexToBeVisited); 
 
-	while (!q.empty()) { 
+	while (!Queue.empty()) {
```
Comment (Bhavnaharitsa) [inline Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: Queue smaller

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp (hunk 6 above)
Comment (akashksinghal) [reply Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: Done

File: Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp (hunk 7 above)
Comment (akashksinghal) [reply Algorithms/Graphs/Detect_Cycle_In_Undirected_Graph.cpp]: Done

**Comment by akashksinghal** [comment]
@Bhavnaharitsa Mam, score for this is not updated on the leaderboard.

**Comment by Bhavnaharitsa** [comment]
Scores for only 2 prs are counted

**Pull request merged**